### PR TITLE
fix: remove reliance on transaction decode in confirmations

### PR DESCRIPTION
**Pull request by matthewwalsh0**
## **Description**

Disable all advanced transaction data decoding using Sourcify, 4Byte and Uniswap, if the `Decode smart contracts` toggle is disabled.

Remove all reliance on the advanced decoding excluding the `Data` section.

Specifically: 

- Create `useTokenTransactionData` hook to decode all token transactions locally using the ABIs.
- Replace all usages of `useDecodedTransactionData` with the new hook, except for the `TransactionData` component.
- Update related unit and integration tests to decode valid test data rather than rely on mocking hooks. 

[Open in GitHub Codespaces](https://codespaces.new/MetaMask/metamask-extension/pull/29341?quickstart=1)

## **Manual testing steps**

Regression of redesigned transaction confirmations.

Specifically:

- Token Transfer Recipient
- Token Transfer Amount
- Approval Spender
- Approval Spending Cap

### **After**

New Toggle Description

## **Pre-merge author checklist**

- [x] I've followed [MetaMask Contributor Docs](https://github.com/MetaMask/contributor-docs) and [MetaMask Extension Coding Standards](https://github.com/MetaMask/metamask-extension/blob/main/.github/guidelines/CODING_GUIDELINES.md).
- [x] I've completed the PR template to the best of my ability
- [x] I’ve included tests if applicable
- [x] I’ve documented my code using [JSDoc](https://jsdoc.app/) format if applicable
- [x] I’ve applied the right labels on the PR (see [labeling guidelines](https://github.com/MetaMask/metamask-extension/blob/main/.github/guidelines/LABELING_GUIDELINES.md)). Not required for external contributors.

## **Pre-merge reviewer checklist**

- [ ] I've manually tested the PR (e.g. pull and build branch, run the app, test code being changed).
- [ ] I confirm that this PR addresses all acceptance criteria described in the ticket it closes and includes the necessary testing evidence such as recordings and or screenshots.

### CODE REVIEW

**Review by OGPoyraz — DISMISSED** [review]

File: app/_locales/en/messages.json [hunk 1]
```diff
@@ -6636,7 +6636,7 @@
     "message": "Decode smart contracts"
   },
   "use4ByteResolutionDescription": {
-    "message": "To improve user experience, we customize the activity tab with messages based on the smart contracts you interact with. MetaMask uses a service called 4byte.directory to decode data and show you a version of a smart contract that's easier to read. This helps reduce your chances of approving malicious smart contract actions, but can result in your IP address being shared."
```
Comment (pedronfigueiredo) [inline app/_locales/en/messages.json:6639]: I thought that in order for the new message to be picked up correctly by the translation program (and in order to eventually be translated to other languages) you needed to create a new translation key and delete the old one, including deleting the old one on all languages. I might be wrong though, worth checking with the extension platform team.

**Review by pedronfigueiredo — DISMISSED** [review]

File: ui/pages/confirmations/components/confirm/info/hooks/useDecodedTransactionData.ts [hunk 2]
```diff
@@ -23,6 +26,7 @@ export function useDecodedTransactionData(
 
   return useAsyncResult(async () => {
     if (
+      !isDecodeEnabled ||
```
Comment (danjm) [inline ui/pages/confirmations/components/confirm/info/hooks/useDecodedTransactionData.ts:29]: For the sake of immediately resolving the privacy regression, is there anything more needed than just the above two lines?

Comment (matthewwalsh0) [reply ui/pages/confirmations/components/confirm/info/hooks/useDecodedTransactionData.ts:29]: Afraid so, the redesigned confirmations are reliant on the advanced decoding meaning they would lose core functionality if it were disabled.

File: ui/pages/confirmations/components/confirm/info/shared/send-heading/send-heading.tsx [hunk 3]
```diff
@@ -89,10 +87,6 @@ const SendHeading = () => {
 
   useSendingValueMetric({ transactionMeta, fiatValue });
 
-  if (pending) {
-    return <ConfirmLoader />;
-  }
-
```
Comment (jpuri) [inline ui/pages/confirmations/components/confirm/info/shared/send-heading/send-heading.tsx:95]: 👍

**Review by jpuri — DISMISSED** [review]
Great work

File: app/_locales/en/messages.json (hunk 1 above)
Comment (OGPoyraz) [reply app/_locales/en/messages.json:6639]: That's right, this key should be deleted and new one must be used.

Comment (matthewwalsh0) [reply app/_locales/en/messages.json:6636]: Done.

**Review by pedronfigueiredo — APPROVED** [review]

**Review by OGPoyraz — APPROVED** [review]

File: ui/pages/confirmations/components/confirm/info/hooks/useDecodedTransactionData.ts (hunk 2 above)
Comment (matthewwalsh0) [reply ui/pages/confirmations/components/confirm/info/hooks/useDecodedTransactionData.ts:29]: Afraid so, fundamental data such as token transfer recipient and approve spending cap.

**Review by darkwing — APPROVED** [review]

**Pull request merged**